src/assets/scripts/client/scope/RadarTargetModel.js

```diff
@@ -0,0 +1,411 @@
+         */
+        this._routeString = '';
+
+        /**
+         * A 3 character (or less) alphanumeric string that is shown in the data block
+         * The scratchpad is used for controller shorthand notes and other purposes
+         *
+         * @for RadarTargetModel
+         * @property _scratchPadText
+         * @type {string}
+         */
+        this._scratchPadText = '';
+
+        /**
+         * Active theme
+         *
+         * @for RadarTargetModel
+         * @property _theme
+         * @type {object}
+         */
+        this._theme = theme;
+
+        this._init(aircraftModel);
+    }
+
+    /**
+     * Angle away from the radar target to draw the leader line and data block
+     *
+     * @for RadarTargetModel
+     * @property dataBlockLeaderDirection
+     * @type {number}
+     */
+    get dataBlockLeaderDirection() {
+        return this._dataBlockLeaderDirection;
+    }
+
+    /**
+     * Length of leader line connecting radar target and data block
+     *
+     * @for RadarTargetModel
+     * @property dataBlockLeaderLength
+     * @type {number}
+     */
+    get dataBlockLeaderLength() {
+        return this._dataBlockLeaderLength;
+    }
+
+    /**
+     * Get the `PositionModel` for the aircraft associated with the radar target
+     *
+     * @for RadarTargetModel
+     * @property positionModel
+     * @type {PositionModel}
+     */
+    get positionModel() {
+        return this.aircraftModel.positionModel;
+    }
+
+    /**
+     * Get the latest known altitude for the aircraft associated with the radar target
+     *
+     * @for RadarTargetModel
+     * @property positionModel
+     * @type {PositionModel}
+     */
+    get indicatedAltitude() {
+        return this.aircraftModel.altitude;
+    }
+
+    /**
+     * Complete initialization tasks
+     *
+     * @for RadarTargetModel
+     * @method _init
+     * @param theme {object}
+     * @param aircraftModel {AircraftModel}
+     */
+    _init(aircraftModel) {
+        this.aircraftModel = aircraftModel;
+        this._cruiseAltitude = aircraftModel.fms.flightPlanAltitude;
+        this._dataBlockLeaderDirection = this._theme.DATA_BLOCK.LEADER_DIRECTION;
+        this._dataBlockLeaderLength = this._theme.DATA_BLOCK.LEADER_LENGTH;
+        // TODO: This getter doesn't give us what we want. Seems like one does
+        // not exist actually. We want the full route string, including past legs.
+        this._routeString = aircraftModel.fms.currentRoute;
+
+        this._initializeScratchPad();
+    }
+
+    /**
+     * Assign a new "hard" altitude
+     *
+     * @for RadarTargetModel
+     * @param altitude {number}
+     * @return {array} [success of operation, system's response]
+     */
+    amendAltitude(altitude) {
+        this._cruiseAltitude = altitude;
+
+        return [true, 'AMEND ALTITUDE'];
+    }
+
+    /**
+     * Enable handlers
+     *
+     * @for RadarTargetModel
+     * @method disable
+     */
+    disable() {
```

Comment:
move to after `_init()` with other lifecycle methods defined in calling order

Reply:
done